pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTableRestletResource.java

```diff
@@ -210,21 +214,60 @@ public String listTableConfigs(@ApiParam(value = "realtime|offline") @QueryParam
 
       if (tableType == null) {
         tableNames = _pinotHelixResourceManager.getAllRawTables();
+      } else if (tableType == TableType.REALTIME) {
+        tableNames = _pinotHelixResourceManager.getAllRealtimeTables();
       } else {
-        if (tableType == TableType.REALTIME) {
-          tableNames = _pinotHelixResourceManager.getAllRealtimeTables();
-        } else {
-          tableNames = _pinotHelixResourceManager.getAllOfflineTables();
-        }
+        tableNames = _pinotHelixResourceManager.getAllOfflineTables();
+      }
+      if (sortType == null) {
+        return JsonUtils.newObjectNode().set("tables", JsonUtils.objectToJsonNode(tableNames)).toString();
       }
 
-      Collections.sort(tableNames);
+      if (sortType == "name") {
+        // Sort table names alphabetically
+        Collections.sort(tableNames, sortAsc ? null : Collections.reverseOrder());
+      } else {
+        // Sort table names based on (1) Create Time or (2) Last Modified Time
```

Comment:
I feel it is cleaner to read all table stats first, then sort on them per the sort type. We need to add `tableName` to the table stats in order to make it work.
We should make branches for `creationTime` and `lastModifiedTime`, and throw `IllegalArgumentException` when `sortType` is not valid

Reply:
Shall I add a param for `tableName` in the tableStats? This I would think, make the sorting much more cleaner.

Reply:
Yes, let's add `tableName` to `TableStats`

Reply:
I have added a param for `tableName`, please check if it makes sense.